File: TableViewCellReuseTest/TableViewCellReuseTest/ViewController.m
//
//  ViewController.m
//  TableViewCellReuseTest
//
//  Created by ari 李 on 25/01/2018.
//  Copyright © 2018 ari 李. All rights reserved.
//

#import "ViewController.h"
#import "NSString+ChineseCharactersToSpelling.h"
#import "NSString+PinYin.h"
@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    [self testZH2PY];
}

- (void)testZH2PY {
    NSString* zh = @"重庆";
    KSLog(@"%f",[[NSDate date] timeIntervalSinceReferenceDate]);
    NSString* py = [NSString lowercaseSpellingWithChineseCharacters:zh];
    KSLog(@"%f",[[NSDate date] timeIntervalSinceReferenceDate]);
    py = [zh getFirstLetter];
    KSLog(@"%f",[[NSDate date] timeIntervalSinceReferenceDate]);
    KSLog(@"%@",py);
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: UIKit/UILabel/UILabel/ViewController.swift
//
//  ViewController.swift
//  UILabel
//
//  Created by ari 李 on 11/01/2018.
//  Copyright © 2018 ari 李. All rights reserved.
//

import UIKit
import SnapKit
class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        sizeFitTest()
        
    }
    
    private func sizeFitTest() {
        let label  = UILabel()
        label.text = "sizeFitTestsizeFitTestsizeFitTestsizeFitTestsizeFitTestsizeFitTest"
        view.addSubview(label)
        
        label.snp.makeConstraints { (maker) in
            maker.left.equalToSuperview().offset(15)
            maker.right.equalToSuperview().inset(15)
            maker.top.equalTo(view.safeAreaLayoutGuide.snp.top).offset(10)
        }
        let size = label.sizeThatFits(CGSize(width: 200, height: CGFloat(MAXFLOAT)))
        printLog(size)
        
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

